pyseir/ensembles/ensemble_runner_v2.py

```diff
@@ -0,0 +1,237 @@
+    """
+    Compute the peak times for each compartment by finding the arg
+    max, and selecting the corresponding time.
+
+    Parameters
+    ----------
+    value_stack: array[n_samples, time steps]
+        Array with the stacked model output results.
+    t_list: array
+        Array of timesteps.
+
+    Returns
+    -------
+    peak_data: dict
+        For each confidence interval, produce key, value pairs for e.g.
+            - peak_time_cl50
+            - peak_value_cl50
+        Also add peak_value_mean.
+    """
+    peak_indices = value_stack.argmax(axis=1)
+    peak_times = [t_list[peak_index] for peak_index in peak_indices]
+    values_at_peak_index = [val[idx] for val, idx in zip(value_stack, peak_indices)]
+
+    peak_data = dict()
+    output_percentiles = [5, 25, 32, 50, 75, 68, 95]
+    for percentile in output_percentiles:
+        peak_data[f"peak_value_ci{percentile}"] = np.percentile(
+            values_at_peak_index, percentile
+        ).tolist()
+        peak_data["peak_time_ci{percentile}"] = np.percentile(peak_times, percentile).tolist()
+
+    peak_data["peak_value_mean"] = np.mean(values_at_peak_index).tolist()
+    return peak_data
+
+
+def _generate_output_for_suppression_policy(model_ensemble):
+    """
+    Generate output data for a given suppression policy.
+
+    Parameters
+    ----------
+    model_ensemble: list(SEIRModel)
+        List of models to compute the surge windows for.
+
+    Returns
+    -------
+    outputs: dict
+        Output data for this suppression policc ensemble.
+    """
+    outputs = defaultdict(dict)
+    outputs["t_list"] = model_ensemble[0].t_list.tolist()
+
+    # ------------------------------------------
+    # Calculate Confidence Intervals and Peaks
+    # ------------------------------------------
+    for compartment, value_stack in _generate_compartment_arrays(model_ensemble).items():
+        compartment_output = dict()
+
+        # Compute percentiles over the ensemble
+        output_percentiles = [5, 25, 32, 50, 75, 68, 95]
+        for percentile in output_percentiles:
+            outputs[compartment][f"ci_{percentile}"] = np.percentile(
+                value_stack, percentile, axis=0
+            ).tolist()
+
+        if compartment in compartment_to_capacity_attr_map:
+            (
+                compartment_output["surge_start"],
+                compartment_output["surge_start"],
+            ) = _get_surge_window(model_ensemble, compartment)
+            compartment_output["capacity"] = [
+                getattr(m, compartment_to_capacity_attr_map[compartment]) for m in model_ensemble
+            ]
+
+        compartment_output.update(_detect_peak_time_and_value(value_stack, outputs["t_list"]))
+
+        # Merge this dictionary into the suppression level one.
+        outputs[compartment].update(compartment_output)
+
+    return outputs
+
+
+if __name__ == "__main__":
+    print("Running Main")
+    example_inputs = {
+        "fips": "16",
```

Comment:
would prefer to separate out fips from the rest of the inputs.  

And could you make the other inputs a dataclass? 😍